### PR TITLE
[Artifacts] Fix artifact identifier resolution

### PR DESCRIPTION
**Pull request by TomerShor**
When tagging artifacts, we create an `ArtifactIdentifier` with values extracted from the artifact objects, and send it to the API.
The key field of the identifier extracted the artifact's `metadata.key` instead of `spec.db_key`, so it failed to find the artifacts to tag unless they both fields were identical (the fields can be identical if the artifact was logged by a project, but not by a context [run]).

Additionally, in the BE we don't need to lock the db if when tagging artifacts, if there are not artifacts to tag.

https://iguazio.atlassian.net/browse/ML-7320

### CODE REVIEW

File: server/api/db/sqldb/db.py [hunk 1]
```diff
@@ -991,6 +991,15 @@ def tag_artifacts(
...
     ):
         artifacts_keys = [artifact.key for artifact in artifacts]
+        if not artifacts_keys:
+            logger.debug(
+                "No artifacts to tag",
+                project=project,
+                tag=tag_name,
+                artifacts=artifacts,
+            )
+            return
```
Comment (liranbg) [inline server/api/db/sqldb/db.py]: can that be in a separate pr?

File: mlrun/db/base.py [hunk 2]
```diff
@@ -242,7 +242,7 @@ def _resolve_artifacts_to_tag_objects(
             )
             artifact_identifiers.append(
                 mlrun.common.schemas.ArtifactIdentifier(
-                    key=mlrun.utils.get_in_artifact(artifact_obj, "key"),
+                    key=mlrun.utils.get_in_artifact(artifact_obj, "db_key"),
```
Comment (liranbg) [inline mlrun/db/base.py:246]: worth a comment on code why do you extract the db_key and not the key or even (logged by project vs logged by run)

**Review by liranbg — CHANGES_REQUESTED** [review]

File: server/api/db/sqldb/db.py (hunk 1 above)
Comment (TomerShor) [reply server/api/db/sqldb/db.py]: You got it https://github.com/mlrun/mlrun/pull/6066

**Review by liranbg — APPROVED** [review]
